src/azul/drs.py

```diff
@@ -97,7 +98,7 @@ class DRSURI(metaclass=ABCMeta):
     @classmethod
     def parse(cls, drs_uri: str) -> 'DRSURI':
```

Comment:
```suggestion
    def parse(cls, drs_uri: str) -> Self:
```

Reply:
I'm not convinced `Self` is appropriate here, although I admit it's not a straightforward case. [The docs say](https://docs.python.org/3/library/typing.html):

> You should not use Self as the return annotation if the method is not guaranteed to return an instance of a subclass when the class is subclassed

I.e., we could/should use `Self` if any only if `issubclass(C, DRSURI)` implies `type(C.parse(...)) is C`. This is currently the case, but only because all existing subclasses override the `parse` method method. If we ever added a 3rd subclass of `DRSURI` that didn't override `parse`, we'd be failing to satisfy the quoted condition.

Reply:
I did find a place where I broke this rule on this PR, but I chose to fix the return value instead of the annotation.